spring-integration-webflux/src/main/java/org/springframework/integration/webflux/outbound/WebFluxRequestExecutingMessageHandler.java

```diff
@@ -169,13 +188,35 @@ protected Object exchange(Supplier<URI> uriSupplier, HttpMethod httpMethod, Http
 										ResponseEntity.status(response.statusCode())
 												.headers(response.headers().asHttpHeaders());
 
-								Mono<?> bodyMono = Mono.empty();
-
-								if (expectedResponseType instanceof ParameterizedTypeReference<?>) {
-									bodyMono = response.body(BodyExtractors.toMono((ParameterizedTypeReference<?>) expectedResponseType));
+								Mono<?> bodyMono;
+
+								if (expectedResponseType != null) {
+									if (this.replyToFlux) {
```

Comment:
Perhaps the property `replyToFlux` should be named `replyAsFlux` or better `responseAsFlux` that's more in line with `responseType`.

Reply:
ON 2nd thought, this property seems redundant. If you set `responseType=ParameterizedTypeReference<Flux<T>>`, what'd you expect to get back?

Reply:
No, that's not. I accept neither options. I try to be aligned here with `WebClient` API.

So, we have there these in the `ClientResponse`:
```
<T> T body(BodyExtractor<T, ? super ClientHttpResponse> extractor);

<T> Mono<T> bodyToMono(Class<? extends T> elementClass);

<T> Mono<T> bodyToMono(ParameterizedTypeReference<T> typeReference);

<T> Flux<T> bodyToFlux(Class<? extends T> elementClass);

<T> Flux<T> bodyToFlux(ParameterizedTypeReference<T> typeReference);
```

So, `expectedResponseType` is indeed just a type for the response element, meanwhile `toMono/toFlux` is an API level, so that would be a bit confusing for end-user to say that to select `flux/mono` API, please, specify the particular type. And I named it `toFlux` definitely after that `bodyToFlux` API. Again - for consistency.

This is indeed not about `response`. That is what we will have as a `payload` of reply message - the result of `bodyToMono()` or `bodyToFlux()` methods invocation underneath.

I am also against checking the type for the `ClientResponse`. As I mentioned we have that ` body(BodyExtractor<T, ? super ClientHttpResponse> extractor);` API on the `WebClient` level, so that should be very clear and consistent if we provide a hook to inject any customization  to the `ClientResponse` via `BodyExtractor`. Right, we might call it like `IdentityBodyExtractor` instead, without any low-level operation but just return for the `ClientHttpResponse`. And that is how we will skip any conversion underneath.

Reply:
Right, but since Spring Integration WebFlux is fully based on the Srping WebFlux that really would be better to stay consistence with their API. So, we are going to use existing `BodyExtractor` strategy as an option to inject into this gateway. But at the same time since there is no such a "void" implementation in the Spring WebFlux for us, we are going to place it exactly in the `org.springframework.integration.webflux.support`.

Right, we will also talk to Spring team about having that impl there, in their package, but I'm afraid they just answer to us with an argument that they have already `requestSpec.exchange()` which returns exactly `Mono<ClientResponse>` for your consideration. 

Since we don't have such a luxury here in SI to cover all the `WebClient` API, I think like middle ground with the `BodyExtractor` injection and some `IdentityBodyExtractor` as a point to get access to the ClientResponse` as is, should be enough to cover as much as possible from your original request.

Sounds good ?

Reply:
Sounds good to me. Thanks for making the PR.